### DISCUSSION TITLE
record comparison of more than two input datasets

### DISCUSSION THREAD

**Discussion by 1tuanh1** [moj-analytical-services/splink#1537] [Q&A] [answered]
Hi,
How can I link e.g. three (or even more) input datasets and how does blocking work in this case? 
I couldn't find any information in the documentation besides  "the other options are link_only, or link_and_dedupe, which may be used if there are multiple input datasets" https://moj-analytical-services.github.io/splink/demos/tutorials/04_Estimating_model_parameters.html?h=link_a 

Can you provide an example?

Really appreciate your help! :)

**Comment by RobinL** [comment, marked as answer]
On my phone so sorry for lack of detail, but fundamentally all that changes is you pass in three dataframes  to the linker:


`linker = DuckDBLinker([df1, df2, df3], settings)`

In the link_only case, only between-dataframe comparisons are created, according to your blocking rules.  For link_and_desupe, it will create all within and between dataframe comparisons, according to your blocking rules

**Reply by aflaxman** [reply]
I made a little example with `pseudopeople` to see how this works, you can find it here: https://colab.research.google.com/drive/19eV-lT_ZvFPMhf7gO9iwmF0MfeYBPGli?usp=sharing

**Reply by 1tuanh1** [reply]
Many thanks to both of you! :)
